src/include/repository.coffee

```diff
@@ -496,17 +496,23 @@ module.exports=
     name: 'punctuation.separator.comma.haskell'
     match: /,/
   lit_num: [
+    name: 'constant.numeric.hexfloat.haskell'
+    match: '0[xX]_*[0-9a-fA-F](_*[0-9a-fA-F])*((\\.[0-9a-fA-F](_*[0-9a-fA-F])*(_*[pP][+-]?[0-9](_*[0-9])*)*)|(_*[pP][+-]?[0-9](_*[0-9])*))'
```

Comment:
This is rather hard to read frankly, and I'm not a fan of duplication. Abstracting general "floating point" pattern into a function accepting a character class could be a good idea (as far as I can tell, character class is the only difference)

Reply:
I will comment together in the next place.